airbyte-tests/src/automaticMigrationAcceptanceTest/java/io/airbyte/test/automaticMigrationAcceptance/MigrationAcceptanceTest.java

```diff
@@ -28,51 +28,66 @@
 import java.nio.file.Path;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.time.Duration;
 import java.util.List;
-import java.util.Map;
 import java.util.Properties;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
+import java.util.function.Supplier;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * In order to run this test from intellij, build the docker images via SUB_BUILD=PLATFORM ./gradlew
- * composeBuild and replace System.getenv("MIGRATION_TEST_VERSION") with the version in your .env
- * file
+ * composeBuild and set VERSION in .env to the pertinent version.
  */
 public class MigrationAcceptanceTest {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MigrationAcceptanceTest.class);
 
   // assume env file is one directory level up from airbyte-tests.
-  private final static File ENV_FILE = Path.of(System.getProperty("user.dir")).getParent().resolve(".env").toFile();
+  private static final File ENV_FILE = Path.of(System.getProperty("user.dir")).getParent().resolve(".env").toFile();
+
+  private static final String TEST_DATA_DOCKER_MOUNT = "airbyte_data_migration_test";
+  private static final String TEST_DB_DOCKER_MOUNT = "airbyte_db_migration_test";
+  private static final String TEST_WORKSPACE_DOCKER_MOUNT = "airbyte_workspace_migration_test";
+  private static final String TEST_LOCAL_ROOT = "/tmp/airbyte_local_migration_test";
+  private static final String TEST_LOCAL_DOCKER_MOUNT = "/tmp/airbyte_local_migration_test";
 
   @Test
   public void testAutomaticMigration() throws Exception {
-    // default to version in env file but can override it.
-    final String targetVersion;
-    if (System.getenv("MIGRATION_TEST_VERSION") != null) {
-      targetVersion = System.getenv("MIGRATION_TEST_VERSION");
-    } else {
-      final Properties prop = new Properties();
-      prop.load(new FileInputStream(ENV_FILE));
-      targetVersion = prop.getProperty("VERSION");
-    }
-    LOGGER.info("Using version: {} as target version", targetVersion);
+    // run version 17 (oldest version of airbyte that supports auto migration)
+    final File version17DockerComposeFile = MoreResources.readResourceAsFile("docker-compose-migration-test-0-17-0-alpha.yaml");
+    final Properties version17EnvVariables = MoreProperties
+        .envFileToProperties(MoreResources.readResourceAsFile("env-file-migration-test-0-17-0.env"));
+    runAirbyte(version17DockerComposeFile, version17EnvVariables, () -> {
+      populateDataForFirstRun();
+      healthCheck(getApiClient());
+    });
+
+    // attempt to run from pre-version bump version to post-version bump version. expect failure.
+    final File currentDockerComposeFile = MoreResources.readResourceAsFile("docker-compose.yaml");
+    final Properties envFileProperties = overrideDirectoriesForTest(MoreProperties.envFileToProperties(ENV_FILE));
```

Comment:
in the env files that are "preserved", we write the values directly into them. but since our goal for migrating to the current version is to piggyback off of whatever the existing env file is, we have to do this override right at the end.

Reply:
adding a comment to explain.

Reply:
That makes sense. But it spawned a new question: why don't we need to call this for the final migration that this test performs, to the current version?

Reply:
we do. it reuses the same variable.

Reply:
Ah you're right, my mistake. Looks good 👍